integrations-service/integrations/utils/integrations/mailgun.py

```diff
@@ -0,0 +1,63 @@
+import aiohttp
+from beartype import beartype
+from tenacity import retry, stop_after_attempt, wait_exponential
+
+from ...autogen.Tools import MailgunSendEmailArguments, MailgunSetup
+from ...env import mailgun_api_key  # Import env to access environment variables
+from ...models import MailgunSendEmailOutput
+
+
+@beartype
+@retry(
+    wait=wait_exponential(multiplier=1, min=4, max=10),
+    reraise=True,
+    stop=stop_after_attempt(3),
+)
+async def send_email(
+    setup: MailgunSetup, arguments: MailgunSendEmailArguments
+) -> MailgunSendEmailOutput:
+    """
+    Sends an email with the provided details using the Mailgun API.
+    """
+
+    try:
+        domain = arguments.from_.split("@")[-1]
+    except Exception as e:
```

Comment:
could you refactor this a bit? I don't think there will be any error raised here

Reply:
you're right. done.